### PR TITLE
PLAT-117042: Fix VirtualList to call `onScrollStop` when scrollbar's visibility changed while scrolling

### PR DESCRIPTION
**Pull request by MikyungKim**
Enact-DCO-1.0-Signed-off-by: Mikyung Kim (mikyung27.kim@lge.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [ ] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
When the visibility of scrollbars changed, current `scrollStopJob` stopped and we create `scrollStopJob` to reference the recent function. But if `scrollStopJob` is already started and did not call its function, some internal status may not be correct. In PLAT-117042, `scrolling` did not reset.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Checked if the internal status `scrolling` is true then call `run` before `stop` it.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
In Scrollbar component, sometimes an error that showing `scrollbarTrackRef.current` is null. So I've added a guard for it.

### Links
[//]: # (Related issues, references)
PLAT-117042

### CODE REVIEW

File: packages/ui/useScroll/Scrollbar.js [hunk 1]
```diff
@@ -50,7 +50,9 @@ const useScrollbar = (props) => {
 	hideScrollbarTrackJob.current = hideScrollbarTrackJob.current || new Job(hideScrollbarTrack, scrollbarTrackHidingDelay);
 
 	function hideScrollbarTrack () {
-		removeClass(scrollbarTrackRef.current, css.scrollbarTrackShown);
+		if (scrollbarTrackRef.current !== null) {
+			removeClass(scrollbarTrackRef.current, css.scrollbarTrackShown);
```
Comment (ybsung) [inline packages/ui/useScroll/Scrollbar.js:54]: How about moving the condition checking `null` into the `removeClass`? If the `removeClass` is called several times, we need to wrap it with condition several times with your code.

File: packages/ui/useScroll/useScroll.js [hunk 2]
```diff
@@ -392,6 +392,9 @@ const useScrollBase = (props) => {
 		}
 
 		return () => {
+			if (ref.scrollStopJob.alive()) {
+				ref.scrollStopJob.run();
+			}
```
Comment (ybsung) [inline packages/ui/useScroll/useScroll.js:397]: There is another scrollStopJob.stop() call in the following code. If this code works first, then the `ref.scrollStopJob.alive()` would be false and it works incorrectly. Am I right?

```
	useEffect(() => {
		mutableRef.current.resizeRegistry.parent = context;

		// componentWillUnmount
		return () => {
			const {... scrollStopJob} = mutableRef.current;

			...
			scrollStopJob.stop();
```

Comment (ybsung) [reply packages/ui/useScroll/useScroll.js:397]: I got it! Their lifecycles are different.

**Review by ybsung — APPROVED** [review]
LGTM

File: packages/core/util/Job.js [hunk 3]
```diff
@@ -25,6 +25,8 @@ class Job {
 		this.timeout = timeout;
 	}
 
+	alive = () => Boolean(this.id);
```
Comment (MikyungKim) [inline packages/core/util/Job.js:28]: @ryanjduffy I'm going to add this function in Job. WDYT?

File: packages/ui/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,12 @@
 
 The following is a curated list of changes in the Enact ui module, newest changes on the top.
 
+## [unreleased]
+
+### Fixed
+
+- `ui/Scroller` and `ui/VirtualList` to call remaining scrollStopJob function in cleanup phase
```
[the remark this answers is not in the document]
Comment (MikyungKim) [reply packages/ui/CHANGELOG.md:9]: Actually the bug itself is from theme(sandstone) side not ui side.
Can I mention the theme-side bug here then?
e.g. Fixed `ui/VirtualList` to not lose focus when scrollbar visibility has changed

File: packages/core/util/Job.js (hunk 3 above)
Comment (MikyungKim) [reply packages/core/util/Job.js:28]: I see your point. I'll try the first suggestion of yours. Thanks!

**Comment by 0x64** [comment]
Looks sound!

**Pull request merged**